### PR TITLE
Fix submitter

### PR DESCRIPTION
**Pull request by lightsing**

### CODE REVIEW

File: src/tele_out/task_fetcher.rs [hunk 1]
```diff
@@ -34,64 +32,52 @@ impl TaskFetcher {
...
-                inner join {} b on t.block_id = b.block_id
-            where b.status = $1 and t.status = $2
-            ORDER BY t.block_id ASC
-            LIMIT 1"#,
+                     inner join {} l2b
+                                on t.block_id = l2b.block_id
+            where t.block_id < coalesce((select block_id
+                                         from task
+                                         where status <> 'proved'
+                                         order by block_id
```
Comment (HAOYUatHZ) [inline src/tele_out/task_fetcher.rs:56]: `order by block_id DESC`?   

or do we really need this ` t.block_id > XXX`?

File: src/tele_out/task_fetcher.rs [hunk 2]
```diff
@@ -34,64 +32,52 @@ impl TaskFetcher {
...
+                                         limit 1), 0)
+              and t.block_id > $1
+              and t.status = 'proved' -- defense filter
+              and l2b.status = 'uncommited'
+            order by t.block_id
```
Comment (HAOYUatHZ) [inline src/tele_out/task_fetcher.rs:61]: better to make ` order by t.block_id ASC` explicitly?

Comment (lightsing) [reply src/tele_out/task_fetcher.rs:61]: asc is always the default order, this is redundant

File: src/tele_out/task_fetcher.rs (hunk 1 above)
Comment (lightsing) [reply src/tele_out/task_fetcher.rs:56]: this select the first unproved block to limit the range of submit.
`and t.block_id > $1` limit the range to un-submitted blocks during running.

**Review by HAOYUatHZ — APPROVED** [review]

**Pull request merged**